src/main/java/com/mtb/demo/dto/ProductDTO.java

```diff
@@ -1,25 +1,62 @@
 package com.mtb.demo.dto;
 
+
 import java.util.Collection;
 import java.util.stream.Collectors;
-
+import com.mtb.demo.entity.Gender;
+import lombok.AllArgsConstructor;
 import lombok.Builder;
 import lombok.Getter;
+import lombok.NoArgsConstructor;
+
 
 @Builder
 @Getter
+@AllArgsConstructor
 public class ProductDTO {
 
+
 	private final String name;
 	private final Collection<SizeDto> sizes;
+	private final Collection<BrandDto> brandNames;
+	private final Collection<ColorDto> colors;
+	private final Collection<GenderDto> genders;
+
 
 	/**
 	 * used only for representation in thymeleaf
 	 * @return
 	 */
 	public String toSizesStr() {
+		if (this.sizes == null) {
+			return "";
+		}
 		return sizes.stream().
 				map(SizeDto::getName)
 				.collect(Collectors.joining(", "));
 	}
+	public String toBrandStr() {
+		if (this.brandNames == null) {
+			return "";
+		}
+		return brandNames.stream().
+				map(BrandDto::getBrandName)
+				.collect(Collectors.joining(", "));
+	}
+	public String toColorStr() {
+		if (this.colors == null) {
+			return "";
+		}
+		return colors.stream().
+				map(ColorDto::getColor)
+				.collect(Collectors.joining(", "));
+	}
+	public String toGenderStr() {
+		if (this.genders == null) {
+			return "";
+		}
+		return genders.stream()
+				.map(GenderDto::getCode)
+				.collect(Collectors.joining(", "));
+	}
```

Comment:
minor: separate methods with empty line

Reply:
I see decupled code here. Maybe some generic method will be nice to have in as in ListUtil class.

Reply:
Suggestion what it could be:

```java
public static <T> String joinByCommaListNamesOf(Collection<T> list, Function<T, String> mapper) {
	if (list.isEmpty()) {
		return "";
	}
	
	return list.stream()
			.map(mapper)
			.collect(Collectors.joining(", "));
}
```

and how to use:

```java
return joinByCommaListNamesOf(sizes, SizeDto::getName);
```

Reply:
Cia issukis, bet gerai, kad pavydi davet, PAskui susiradau paskaitos